Tests/AssemblerLib/LocalToGlobalIndexMapMultiComponent.cpp

```diff
@@ -74,52 +74,29 @@ class AssemblerLibLocalToGlobalIndexMapMultiDOFTest : public ::testing::Test
 		    mesh_items_all_nodes->getIntersectionByNodes(nodes));
 	}
 
-	virtual ~AssemblerLibLocalToGlobalIndexMapMultiDOFTest()
-	{
-		for (auto p : components_boundary)
-			delete p;
-		for (auto p : components)
-			delete p;
-		for (auto p : boundary_elements)
-			delete p;
-	}
-
-	void clear()
-	{
-		for (auto* mss : components) {
-			delete mss;
-		}
-
-		components.clear();
-
-		for (auto* mss : components_boundary) {
-			delete mss;
-		}
-
-		components_boundary.clear();
-	}
-
 	void initComponents(const unsigned num_components, const unsigned selected_component,
 						const AL::ComponentOrder order)
 	{
 		assert(selected_component < num_components);
 
-		clear();
-
+		std::vector<std::unique_ptr<MeshLib::MeshSubsets>> components;
 		for (unsigned i=0; i<num_components; ++i)
 		{
-			components.push_back(new MeL::MeshSubsets(
-			    mesh_items_all_nodes.get()));
+			components.emplace_back(std::unique_ptr<MeshLib::MeshSubsets>{
+                    new MeL::MeshSubsets{
```

Comment:
Looks like your shiny clang auto-formatter fails miserably! I always knew, it sucks. What do you say now, eh?
:laughing: